x-pack/platform/plugins/shared/task_manager/server/task_store.ts

```diff
@@ -902,21 +902,22 @@ export class TaskStore {
     { max_docs: max_docs }: UpdateByQueryOpts = {}
   ): Promise<UpdateByQueryResult> {
     const { query } = ensureQueryOnlyReturnsTaskObjects(opts);
+    const { sort, ...rest } = opts;
     try {
-      const // @ts-expect-error elasticsearch@9.0.0 https://github.com/elastic/elasticsearch-js/issues/2584 types complain because the body should not be there.
-        // However, we can't use this API without the body because it fails to claim the tasks.
+      const // However, we can't use this API without the body because it fails to claim the tasks.
         // eslint-disable-next-line @typescript-eslint/naming-convention
         { total, updated, version_conflicts } = await this.esClientWithoutRetries.updateByQuery(
           {
             index: this.index,
             ignore_unavailable: true,
             refresh: true,
             conflicts: 'proceed',
-            body: {
-              ...opts,
-              max_docs,
-              query,
-            },
+            ...rest,
+            max_docs,
+            query,
+            // @ts-expect-error According to the docs, sort should be a comma-separated list of fields and goes in the querystring.
+            // However, this one is using a "body" format?
+            body: { sort },
```

Comment:
@JoshMock, @pquentin, this is an interesting one:
According to the docs, only the `sort` querystring parameter is allowed (and the format is a comma-separated list of fields).
However, this query has always sent `sort` in the body (with an object format).

Reply:
I know there are some APIs that are permissive about putting some query parameters in the body instead, but the spec may be more strict, intentionally or not. @pquentin I know this has come up a couple times recently in the spec; do we have a preference on how the spec handles things that can go in query or body?